flexget/plugins/filter/file_age.py

```diff
@@ -0,0 +1,67 @@
+from __future__ import unicode_literals, division, absolute_import
+import logging
+import os
+import sys
+from datetime import datetime
+
+from flexget import plugin
+from flexget.event import event
+from flexget.utils.tools import parse_timedelta
+
+log = logging.getLogger('file_age')
+
+
+class FileAge(object):
+    """
+        Rejects/accepts files based on file metadata
+
+        Example::
+
+          file_age:
+            stat: 'access'
+            age: '7 days'
+    """
+
+    schema = {
+        'type': 'object',
+        'properties': {
+            'stat': {'type': 'string', 'enum': ['access', 'modified', 'created']},
+            'reject': {'type': 'boolean', 'default': True},
+            'age': {'type': 'string', 'format': 'interval'}
+        },
+        'required': ['stat', 'age'],
+        'additionalProperties': False
+    }
+
+    def on_task_filter(self, task, config):
+        for entry in task.entries:
+            if 'location' not in entry:
+                log.debug('Entry %s is not a local file. Skipping.', entry['title'])
+                continue
+            filepath = entry['location']
+            if config['stat'] == 'created' and not sys.platform.startswith('win'):
+                log.warning('File creation time is not available in *nix OS and defaults to modification time.')
+            file_stats = {}
+            file_stats['access'] = datetime.fromtimestamp(os.path.getatime(filepath))
```

Comment:
Filesystem plugin should probably set these as entry fields. Then the "age" plugin could work nicely like

age:
  field: access
  amount: 7 days
  action: reject

and thus be generic